app/javascript/components/bulk_edit/CaseStatus.js

```diff
@@ -0,0 +1,212 @@
+import React from 'react';
+import { Button, Modal, Form } from 'react-bootstrap';
+import { PropTypes } from 'prop-types';
+import axios from 'axios';
+import reportError from '../util/ReportError';
+
+class CaseStatus extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+      patients: [],
+      case_status: '',
+      message: '',
+      confirmed: '',
+      isolation: undefined,
+      apply_to_group: false,
+      monitoring: false,
+      monitoring_reason: '',
+      public_health_action: '',
+      loading: false,
+    };
+    this.submit = this.submit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.toggleCaseStatusModal = this.toggleCaseStatusModal.bind(this);
+
+    window.bulkEditCaseStatusComponent = this;
+  }
+
+  activate(patients) {
+    if (!patients || !patients.length) {
+      return;
+    }
+
+    this.getCommonAttributes(patients);
+
+    this.setState({
+      showModal: true,
+      patients: patients,
+    });
+  }
+
+  toggleCaseStatusModal() {
+    let current = this.state.showModal;
+    this.setState({
+      showModal: !current,
+    });
+  }
+
+  getCommonAttributes(patients) {
+    const distinctCaseStatus = [...new Set(patients.map(x => x.case_status))];
+    console.log(distinctCaseStatus);
+    if (distinctCaseStatus.length === 1) {
+      this.setState({
+        case_status: distinctCaseStatus[0],
+      });
+    }
+  }
+
+  handleChange(event) {
```

Comment:
Still noticing a few things regarding the messages and options that show up when attempting to do the bulk edit:

1) When bulk editing patients in isolation and changing case status to confirmed/probable, the options include "Continue Monitoring in Isolation Workflow", should probably either only include "End Monitoring" or not have any options at all to change the monitoring status @holmesie should it be the latter, similar to the way it works for the single patient update?

![Screen Shot 2020-06-24 at 8 50 43 AM](https://user-images.githubusercontent.com/9956561/85559988-3beb1380-b5f8-11ea-8186-f06858cbee58.png)

2) Not as big of a deal, but the spacing between the select input and the message in this scenario happens to be different/inconsistent with the other ones

![Screen Shot 2020-06-24 at 8 51 07 AM](https://user-images.githubusercontent.com/9956561/85560390-9a17f680-b5f8-11ea-868c-b93fd17c64c1.png)

3) After going back to exposure monitoring and trying to do a bulk edit, it seems like some kind of previous state was not reset/cleared because it's displaying the "This case will be moved..." without anything being selected yet

![Screen Shot 2020-06-24 at 8 51 44 AM](https://user-images.githubusercontent.com/9956561/85560603-ce8bb280-b5f8-11ea-8de9-b4b22bbd01dd.png)

4) When in isolation and moving patients back to exposure, the message says "This case will be moved from PUI to..." when it should say something along the lines of "This case will be moved from the isolation workflow to..." (I know this is currently the way it is in the single patient edit, but this is something we are planning to fix there as well)

![Screen Shot 2020-06-24 at 8 50 02 AM](https://user-images.githubusercontent.com/9956561/85559940-2ece2480-b5f8-11ea-9cc5-e88289e3215a.png)

5) Since these changes are to be applied to patients in bulk, I think it would make sense for us to change the wording of these messages from "This patient..." to "These patients..."

Reply:
@timwongj The biggest reason for changing case status to confirmed or probable is moving them from exposure to isolation (and vice versa for the other case status options).

Reply:
@holmesie understood, is this in response to my question from #1?

If so, should there just be no follow up options when changing from an isolation case status (confirmed or probable) to another isolation case status?

Reply:
It was, and yes - the followup option should only really be there if "Confirmed" and "Probable" are selected, and that is because we need to figure out if the monitorees need to be closed out, or if they need to be moved from exposure to isolation (also note there is no follow up questions for the reverse case of isolation back to exposure).

Reply:
great, thanks for the clarification!

Reply:
I just pushed a change that will check the initial `isolation` and `monitoring` and if in isolation but open it will only show the end monitoring follow up. If in isolation and closed it does not show any follow up. That should address that issue and reduce any confusion.

Reply:
As is, there should not be any follow up question for isolation back to exposure. In the latest commit, the only time a follow up will be prompted in isolation is in the case that you change for example `Probable` to `Confirmed` but now want to also end monitoring/move to closed. Only then will the `End Monitoring` option appear.

Reply:
ah, so for that case, we currently don't display any options for the single patient edit, so I'm we would want the same behavior for the bulk edit. @holmesie is this correct?

Reply:
I don't think so, I'd rather add an additional "bulk edit" button solely for closing our records regardless of workflow.

Reply:
Okay so if I follow correctly, no follow up when starting in isolation?

Reply:
so when changing from Probable to Confirmed when already in isolation, there should not be an option to close records, as that would eventually be in a separate bulk edit action right?